llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -950,10 +950,13 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
   // promoted to f32. v2f16 is expanded to f16, which is then promoted
   // to f32.
   for (const auto &Op :
-       {ISD::FDIV, ISD::FREM, ISD::FSQRT, ISD::FSIN, ISD::FCOS}) {
+       {ISD::FDIV, ISD::FREM, ISD::FSQRT, ISD::FSIN, ISD::FCOS, ISD::FTANH}) {
     setOperationAction(Op, MVT::f16, Promote);
     setOperationAction(Op, MVT::f32, Legal);
-    setOperationAction(Op, MVT::f64, Legal);
+    // fsin, fcos, and ftanh are not supported on f64
+    if (Op == ISD::FDIV || Op == ISD::FREM || Op == ISD::FSQRT) {
```

Comment:
I'd rephrase the condition or the comment to use matching terms.
I.e. either "if (!(Op == FSIN || Op == FCOS || Op == FTANH))" or the comment should say that "only div/rem/sqrt are legal for f64".

Reply:
Fixed comment